azurerm/internal/services/cdn/cdn_profile_resource.go

```diff
@@ -121,7 +122,12 @@ func resourceArmCdnProfileCreate(d *schema.ResourceData, meta interface{}) error
 		return fmt.Errorf("Cannot read CDN Profile %s (resource group %s) ID", name, resGroup)
 	}
 
-	d.SetId(*read.ID)
+	id, err := parse.CdnProfileID(*read.ID)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(id.ID(subscriptionId))
```

Comment:
we'll also need a state migration to update existing resources - see [the frontdoor migrations](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/services/frontdoor/frontdoor_custom_https_configuration_resource.go#L69-L76) for an example